chaindexing-postgres/src/repo.ts

```diff
@@ -55,6 +55,23 @@ export class PostgresRepo extends Repo<Pool, Conn> {
     // TODO: Implement serial streaming version
   }
 
+  async streamContractAddresses_2(conn: Conn, limit = 10, allowOffset = true) {
+    let currentPage = 0;
+    let offset = 0;
+
+    const getNext = async (): Promise<ContractAddress[] | null> => {
+      if (allowOffset) {
+        offset = limit * currentPage;
+        currentPage += 1;
+      }
```

Comment:
Would it stream if `allowOffset` is `false`. I don't think we need this flag at all:

```suggestion
        offset = limit * currentPage;
        currentPage += 1;
```

Reply:
I was thinking of a situation where the caller perform some update on each iteration and want to fetch what is left.
e.g. if on each iteration, the caller deletes the record. then it may want to fetch again from the beginning. 

but that might not truly the necessary, the caller can still move forward with the stream nonetheless.